src/main/java/org/apache/sling/resourceresolver/impl/mapping/MapEntries.java

```diff
@@ -242,7 +260,16 @@ protected boolean doInit() {
                 try {
                     final Map<String, Map<String, Collection<String>>> loadedMap = this.loadAliases(resolver);
                     this.aliasMapsMap = loadedMap;
-    
+
+                    // warn if there are more than a few defunct aliases
+                    if (conflictingAliases.size() >= MAX_REPORT_DEFUNCT_ALIASES) {
+                        log.warn("There are {} conflicting aliases; excerpt: {}", conflictingAliases.size(), conflictingAliases);
+                    }
+                    if (invalidAliases.size() >= MAX_REPORT_DEFUNCT_ALIASES) {
+                        log.warn("There are {} invalid aliases; excerpt: {}", invalidAliases.size(), invalidAliases);
+                    }
+                    conflictingAliases.clear();
```

Comment:
why do you clear these structures? From what I understood, this code is executed only once for the lifetime of the ResourceResolverFactory, so it's not necessary at all. Nothing against it (easy saved bytes), but a comment would explain why this is done here.